### PR TITLE
finished draft polynomial dissimilarity

### PR DESCRIPTION
**Pull request by Michaellin250**
I finished the draft of polynomial dissimilarity. Any one mind looking at it to make sure it's reasonable?

### CODE REVIEW

File: R/polynomial-dissimilarity.R [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  P_A2 <- similarity_score(A2, k, alpha)
+  
+  dist <- norm(P_A1 - P_A2, type = c("F"))/ nrow(A1)**2
+  
+  # create optional results list
+  results <- list()
+  results[[ "adjacency_matrices" ]] <- c(graph_1, graph_2)
+  results[[ "dist" ]] <- dist
+
+  if(results_list) {
```
Comment (travisbyrum) [inline R/polynomial-dissimilarity.R]: I wouldn't conditionally return the results list.  Just return something like this:

```R
structure(
        list(
          adjacency_matrices = c(graph_1, graph_2),
          dist = dist
        ),
        class = "polynomial_dissimilarity"
)
```

That way you can add s3 methods if you want to do something else with the result.

File: R/polynomial-dissimilarity.R [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  e <- eigen(A)
+  eig_vals <- e$values
+  Q <- e$vectors 
+  n <- nrow(A)
+  polynomial <- function(kp) {
+    eig_vals ** kp / (n - 1) ** (alpha * (kp - 1))
+  }
+  
+  W = Diagonal(sum(sapply(1:k, polynomial)))
+  P_A = dot(dot(Q, W), t(Q))
```
Comment (travisbyrum) [inline R/polynomial-dissimilarity.R]: You don't need to assign this.  Just end the function with `dot(dot(Q, W), t(Q))`.

File: R/polynomial-dissimilarity.R [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+# Calculate the similarity score used in the polynomial dissimilarity distance. 
+similarity_score <- function (A, k, alpha) {
+  
+  e <- eigen(A)
+  eig_vals <- e$values
+  Q <- e$vectors 
+  n <- nrow(A)
+  polynomial <- function(kp) {
```
Comment (travisbyrum) [inline R/polynomial-dissimilarity.R]: I would make this anonymous.

**Review by travisbyrum — CHANGES_REQUESTED** [review]
We need a test case before this goes in.

File: R/polynomial-dissimilarity.R (hunk 3 above)
Comment (Michaellin250) [reply R/polynomial-dissimilarity.R]: Could you clarify what this means? I'm not sure what anonymous means in this example. Do you mean by putting eig_values inside the polynomial function?

Comment (travisbyrum) [reply R/polynomial-dissimilarity.R]: It means writing the `sapply` like this 

```R  
sapply(1:k, function(kp) {
    eig_vals ** kp / (n - 1) ** (alpha * (kp - 1))
})
```

File: R/polynomial-dissimilarity.R [hunk 4]
```diff
@@ -1,5 +1,7 @@
+library(pracma)
```
Comment (travisbyrum) [inline R/polynomial-dissimilarity.R]: Remove the library line.  When using an exported function there are a couple of ways we need to import it. This walks you through how to do it with [roxygen comments](https://laderast.github.io/2019/02/12/package-building-description-namespace/) . You can also just explicitly reference the namespace like `pracma::dot` which is what I usually do.

Once that works you need to explicitly add the import to the `DESCRIPTION` file.  Here's a way to do that automatically with the `usethis` package.

```R
usethis::use_package("pracma")
```

**Review by travisbyrum — CHANGES_REQUESTED** [review]
I pulled in the changes from master so it will build in the ci.  Check the output and you should see a description of the errors.  Once that is done, add your functions to the package documentation.  Just run the following command:

```R
devtools::document()
```

File: R/polynomial-dissimilarity.R (hunk 4 above)
Comment (Michaellin250) [reply R/polynomial-dissimilarity.R]: How do I check the output?

Comment (travisbyrum) [reply R/polynomial-dissimilarity.R]: Click details where it says `Some checks were not successful`.  Alternatively, you can click the check button in RStudio  under the build package to run package checks.  Or you can use:

```R
devtools::check(document=FALSE)
```

**Review by travisbyrum — APPROVED** [review]

**Pull request merged**